datafusion/src/logical_plan/expr.rs

```diff
@@ -508,6 +508,56 @@ impl Expr {
         }
     }
 
+    /// Return if Expr contains Expr::IsNull or Expr::IsNotNull
+    pub fn has_null_expr(&self) -> Result<bool> {
```

Comment:
I'll check this PR out more carefully tomorrow. I still don't fully understand the need for this function (as I don't think any column exprs can be pushed down past the non preserved join side)